src/main/java/codurance/academyfinalboy/backend/actions/CreateActivity.java

```diff
@@ -0,0 +1,28 @@
+package codurance.academyfinalboy.backend.actions;
+
+import codurance.academyfinalboy.backend.model.team.Activity;
+import codurance.academyfinalboy.backend.model.team.ActivityMember;
+import codurance.academyfinalboy.backend.model.team.TeamService;
+import codurance.academyfinalboy.backend.web.controllers.CreateActivityController;
+import java.util.ArrayList;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CreateActivity {
+  private final TeamService teamService;
+
+  public CreateActivity(TeamService teamService) {
+    this.teamService = teamService;
+  }
+
+  public void execute(CreateActivityController.CreateActivityRequest request) {
+    var members =
+        new ArrayList<>(
+            request.members().stream()
+                .map(member -> new ActivityMember(member.fullName()))
+                .toList());
+
+    teamService.addActivity(
+        request.teamId(), new Activity(request.activityName(), members, request.numberOfGroups()));
```

Comment:
this new Activity instantiation and the new ActivityMembers, would it be ok to delegate it to some other collaborator? Just wondering not gonna ask to change this now

Reply:
i think is okay to have it on the action but we can discuss it and see the pro's and con's :)